subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+    if (pool != undefined) {
+        const timestamp = BigInt(new Date().getTime());
+        pool.transactionCount += 1;
+        pool.totalLiquidity = Big(pool.totalLiquidity)
+            // multiplying by 2 to account for base amount being added
+            .add(Big(liquidityAddedEvt.quoteAmount.toString()).mul(2))
+            .toString();
+        pool.calculatedTimestamp = timestamp;
+        pool.blockNumber = BigInt(ctx.block.height);
+
+        // find baseAsset: Following is only valid for dual asset pools
+        const baseAsset = pool.poolAssets
+            .find((asset) => asset.assetId != pool.quoteAssetId);
+        if (baseAsset == undefined) {
+            throw new Error('baseAsset not found');
+        }
+        baseAsset.totalLiquidity += liquidityAddedEvt.baseAmount;
+        baseAsset.calculatedTimestamp = timestamp;
+        baseAsset.blockNumber = BigInt(ctx.block.height);
+        // find quoteAsset
+        const quoteAsset = pool.poolAssets
+            .find((asset) => asset.assetId == pool.quoteAssetId);
+        if (quoteAsset == undefined) {
+            throw new Error('quoteAsset not found');
+        }
+        quoteAsset.totalLiquidity += liquidityAddedEvt.quoteAmount;
+        quoteAsset.calculatedTimestamp = timestamp;
+        quoteAsset.blockNumber = BigInt(ctx.block.height);
+
+        let tx = await get(ctx.store, PabloTransaction, ctx.event.id);
+        if (tx != undefined) {
+            throw new Error("Unexpected transaction in db");
+        }
+        tx = createTransaction(ctx, pool, who,
+            PabloTransactionType.ADD_LIQUIDITY,
+            Big(liquidityAddedEvt.baseAmount.toString())
+                .div(Big(liquidityAddedEvt.quoteAmount.toString())).toString(),
+            BigInt(baseAsset.assetId),
+            liquidityAddedEvt.baseAmount,
+            pool.quoteAssetId,
+            liquidityAddedEvt.quoteAmount);
+
+        await ctx.store.save(pool);
+        await ctx.store.save(baseAsset);
+        await ctx.store.save(quoteAsset);
+        await ctx.store.save(tx);
+    } else {
+        throw new Error("Pool not found");
+    }
+}
+
+interface SwappedEvent {
+    poolId: bigint,
+    who: Uint8Array,
+    baseAsset: bigint,
+    quoteAsset: bigint,
+    baseAmount: bigint,
+    quoteAmount: bigint,
+    fee: bigint
+}
+
+function getSwappedEvent(event: PabloSwappedEvent): SwappedEvent {
+    if (event.isV2100) {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asV2100;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    } else {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asLatest;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    }
+}
+
+export async function processSwappedEvent(ctx: EventHandlerContext, event: PabloSwappedEvent) {
+    console.info('processing SwappedEvent', event);
```

Comment:
Reviewer note: @hussein-aitlahcen please review this part carefully for any mistakes :)